addons/web_unsplash/static/src/js/unsplash_image_widget.js

```diff
@@ -26,47 +26,27 @@ widgetsMedia.ImageWidget.include({
         this._super.apply(this, arguments);
 
         this._unsplash = {
-            isActive: false,
             selectedImages: {},
             isMaxed: false,
             query: false,
             error: false,
+            records: [],
         };
 
-        // TODO improve this
-        //
-        // This is a `hack` to prevent the UnsplashAPI to be destroyed every
-        // time the media dialog is closed. Indeed, UnsplashAPI has a cache
-        // system to recude unsplash call, it is then better to keep its state
-        // to take advantage from it from one media dialog call to another.
-        //
-        // Unsplash API will either be (it's still being discussed):
-        //  * a service (ideally coming with an improvement to not auto load
-        //    the service)
-        //  * initialized in the website_root (trigger_up)
-        if (unsplashAPI === null) {
-            this.unsplashAPI = new UnsplashAPI(this);
-            unsplashAPI = this.unsplashAPI;
-        } else {
-            this.unsplashAPI = unsplashAPI;
-            this.unsplashAPI.setParent(this);
-        }
-    },
-    /**
-     * @override
-     */
-    start: function () {
-        this.$('.o_we_search_icon').replaceWith(core.qweb.render('web_unsplash.dialog.media.search_icon'));
-        return this._super.apply(this, arguments);
+        // Use a single instance of the UnsplashAPI widget as long as the page
+        // is not refreshed. This allows it to keep its cache and not make the
+        // same request many times.
+        unsplashAPI = unsplashAPI || new UnsplashAPI(this);
+        // Need to set parent because RPCs are performed through parent.
+        unsplashAPI.setParent(this);
```

Comment:
Restore previous code and comment, that should still be initialize by the website_root or by a service someday and removing it from `this.` breaks potential custo.